File: Frontend/src/app/product/productType.ts
import { Product } from "./product";

export interface IProductType{
    id: number;
    name: string;
    amount: number;  
    price: number;
 
    product: Product;
}

export class ProductType implements IProductType {     
    constructor(public product: Product,
                public name: string = "Typ",                
                public amount: number = 0,
                public price: number = 0,
                public id: number = 0)
    {
    }

    static AreEqual(a: IProductType, b: IProductType): boolean {   
        return a.product.id == b.product.id
            && a.id == b.id;
    }
}

File: Backend/src/EbecShop.Model/ProductType.cs
using EbecShop.Model.Abstract;
using System;
using System.Collections.Generic;
using System.Text;

namespace EbecShop.Model
{
    public class ProductType : Entity
    {
        public Product Product { get; set; }

        public string Name { get; set; }
        public decimal Price { get; set; }
        public decimal Amount { get; set; }
    }
}
